### PR TITLE
Local rate limit - add rate_limited_as_resource_exhausted flag

### LINKED ISSUE #21461: Support rate_limited_as_resource_exhausted for local rate limit events
*Description*:
- a client service needs to know more details in order to pick the right retry policy, or not retry the call at all.
- a service owner needs to know more details about why a gRPC call failed.

By default, http error 429s are translated to gRPC status `UNAVAILABLE` which can make it difficult to distinguish real downtime from rate limiting events.

Essentially, the same use cases as https://github.com/envoyproxy/envoy/issues/4735.

*Relevant Links*:
PR which added `rate_limited_as_resource_exhausted` for global rate limiting https://github.com/envoyproxy/envoy/pull/4879
https://github.com/grpc/grpc/blob/master/doc/http-grpc-status-mapping.md

### PR DESCRIPTION
**Pull request by Pawan-Bishnoi**
**Commit Message:** Add an option to change grpc status code for rate limited requests
**Additional Description:** The default status code for rate limited requests in Grpc is Unavailable. A more appropriate status code would be ResourceExhausted. We already support this config in Global rate limiting. This PR aims to provide the same option for LRS.

Risk Level: Low
Testing: todo
Docs Changes: todo
Release Notes: todo
Platform Specific Features:
Fixes #21461

### CODE REVIEW

**Comment by Pawan-Bishnoi** [comment]
not stale

**Review by rgs1 — APPROVED** [review]
LGTM, thanks!

**Review by lizan — COMMENTED** [review]
/lgtm api

defer to @wbpcode for code review.

also can you add a release note?

File: test/extensions/filters/http/local_ratelimit/filter_test.cc [hunk 1]
```diff
@@ -45,6 +45,39 @@ stat_prefix: test
       key: x-local-ratelimited
       value: 'true'
 local_rate_limit_per_downstream_connection: {}
+enable_x_ratelimit_headers: {}
+  )";
+static const std::string config_yaml_resource_exhausted = R"(
+stat_prefix: test
+rate_limited_as_resource_exhausted: true
```
Comment (wbpcode) [inline test/extensions/filters/http/local_ratelimit/filter_test.cc:52]: Could we just add a new `rate_limited_as_resource_exhausted: {}` in the original `config_yaml` and use it by setting parameter of `fmt::format` to true

File: test/extensions/filters/http/local_ratelimit/filter_test.cc [hunk 2]
```diff
@@ -145,7 +178,7 @@ TEST_F(FilterTest, RequestOkPerConnection) {
 }
 
 TEST_F(FilterTest, RequestRateLimited) {
-  setup(fmt::format(fmt::runtime(config_yaml), "1", "false", "\"OFF\""));
+  setup(fmt::format(fmt::runtime(config_yaml_resource_exhausted), "1", "false", "\"OFF\""));
```
Comment (wbpcode) [inline test/extensions/filters/http/local_ratelimit/filter_test.cc:181]: could use add a new test case for this new flag?

**Review by wbpcode — CHANGES_REQUESTED** [review]
Thanks for your contribution and some comments are added.

/wait

**Comment by alyssawilk** [comment]
@wbpcode can you PTAL at this?

**Review by wbpcode — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**